src/utils/internal/proxy.ts

```diff
@@ -56,9 +56,13 @@ export function mergeHeaders(
   }
   const merged = new Headers(defaults);
   for (const input of _inputs) {
-    for (const [key, value] of Object.entries(input!)) {
+    const entries =
+      typeof input.entries === "function"
+        ? input.entries()
+        : Object.entries(input);
+    for (const [key, value] of entries) {
       if (value !== undefined) {
-        merged.set(key, value);
+        merged.set(key.toString(), value);
```

Comment:
is `.toString()`  for types only?

Reply:
It's because `Array.entries()` => `[number, string][]`, whereas `Headers.set(name: string, value: string)`

Reply:
might be typescript issue? how keys can be number?